tests/PHPStan/Analyser/data/generics.php

```diff
@@ -844,3 +844,53 @@ public function getFoo()
 	{
 	}
 }
+
+/**
+ * Class Cache
+ *
+ * @template T
+ */
+class Cache
+{
+	/**
+	 * Function Cache::__construct
+	 *
+	 * @param T $t
+	 */
+	public function __construct($t)
+	{
+	}
+
+	/**
+	 * Function Cache::get
+	 *
+	 * @return T
+	 */
+	public function get()
+	{
+	}
+}
+
+/**
+ * Function cache0
+ *
+ * @template T
+ *
+ * @param T $t
+ */
+function cache0($t): void {
+	$c = new Cache($t);
+	assertType('T (function PHPStan\Generics\FunctionsAssertType\cache0(), argument)', $c->get());
+}
+
+/**
+ * Function cache1
+ *
+ * @template T
+ *
+ * @param T $t
+ */
+function cache1($t): void {
+	$c = new Cache($t);
+	assertType('T (function PHPStan\Generics\FunctionsAssertType\cache1(), argument)', $c->get());
```

Comment:
Before the PR, we get `T` from the function `cache0` here, instead of `T` from `cache1`.